src/minimal-live-connect.ts

```diff
@@ -38,22 +38,35 @@ function _minimalInitialization(liveConnectConfig: LiveConnectConfig, externalSt
   }
 }
 
+function _initializeWithoutGlobalName(liveConnectConfig: LiveConnectConfig, externalStorageHandler: ReadOnlyStorageHandler, externalCallHandler: CallHandler, eventBus: EventBus) {
+  const lc = _minimalInitialization(liveConnectConfig, externalStorageHandler, externalCallHandler, eventBus, (event: unknown) => {})
+  window.liQ_instances = window.liQ_instances || []
+  window.liQ_instances.push(lc)
+  return lc
+}
+
+function _initializeWithGlobalName(liveConnectConfig: LiveConnectConfig, externalStorageHandler: ReadOnlyStorageHandler, externalCallHandler: CallHandler, eventBus: EventBus) {
+  const queue = window[liveConnectConfig.globalVarName] = window[liveConnectConfig.globalVarName] || []
+  const push = queue.push.bind(queue)
+  const lc = _minimalInitialization(liveConnectConfig, externalStorageHandler, externalCallHandler, eventBus, push)
+  
+  window.liQ_instances = window.liQ_instances || []
+  if (window.liQ_instances.filter(i => i.config.globalVarName === lc.config.globalVarName).length === 0) {
+    window.liQ_instances.push(lc)
+  }
+  return lc
+}
+
 export function MinimalLiveConnect(liveConnectConfig: LiveConnectConfig, externalStorageHandler: ReadOnlyStorageHandler, externalCallHandler: CallHandler, externalEventBus?: EventBus): ILiveConnect {
+  const configuration = (isObject(liveConnectConfig) && liveConnectConfig) || {}
+  const eventBus = externalEventBus || LocalEventBus()
+  let lc
   try {
-    const configuration = (isObject(liveConnectConfig) && liveConnectConfig) || {}
-    configuration.globalVarName = configuration.globalVarName || 'liQ'
-    if (window) {
-      window[configuration.globalVarName] = window[configuration.globalVarName] || []
-    }
-    const eventBus = externalEventBus || LocalEventBus()
-    const lc = _minimalInitialization(configuration, externalStorageHandler, externalCallHandler, eventBus)
-    window.liQ_instances = window.liQ_instances || []
-    if (window.liQ_instances.filter(i => i.config.globalVarName === configuration.globalVarName).length === 0) {
-      window.liQ_instances.push(lc)
-    }
-    return lc
+    lc = configuration.globalVarName ?
+      _initializeWithGlobalName(configuration, externalStorageHandler, externalCallHandler, eventBus) :
+      _initializeWithoutGlobalName(configuration, externalStorageHandler, externalCallHandler, eventBus)
   } catch (x) {
     console.error(x)
   }
-  return {}
+  return lc
```

Comment:
do we want to keep returning `{}` here, not undefined in case of errors.
Otherwise we will break code like `if (lc.push) ???`

Reply:
Sure. Done in c444ac10d7d88a0521f2af47e50d62383a3706a4